node_modules/gh-series/tests/test-rollover.js

```diff
@@ -0,0 +1,149 @@
+     * every day in every term of several academic years.
+     *
+     * data/rollover-testcases.json is generated by the rollover_testcases
+     * management command from https://github.com/CUL-DigitalServices/timetable-3
+     */
+    it('verify timestamps are rolled over from one academic year to another correctly', function() {
+        // This tests verifies tens of thousands of possible dates, so it can take a while
+        this.timeout(1000 * 10);
+
+        var testData = require('./data/rollover-testcases.json');
+
+        // The rollover code is currently hard-coded to roll over in time local to Europe/London,
+        // so we need to ensure the dataset contains times local to London.
+        assert.strictEqual(testData.roll_zone, 'Europe/London');
+
+        // The years to test rolling between
+        var rolls = _pairwiseCombinations(testData.years);
+
+        // Getting all the possible pairs in an array of length n
+        // results in an array of length: n^2 - n
+        assert.strictEqual(rolls.length, (Math.pow(testData.years.length, 2) - testData.years.length));
+
+        var count = 0;
+        _.each(rolls, function(pair) {
+            var yearFrom = pair[0];
+            var yearTo = pair[1];
+
+            assert.ok(_.isNumber(yearFrom));
+            assert.ok(_.isNumber(yearTo));
+
+            _.each(testData.testcases, function(testcase) {
+                assert.ok(_.isObject(testcase));
+                assert.ok(testcase[yearFrom]);
+                assert.ok(testcase[yearTo]);
+
+                // Timestamps in the test data are UTC
+                assert.strictEqual(moment.tz(testcase[yearFrom], 'UTC').format(), testcase[yearFrom]);
+
+                // Roll over the the "from" date and verify the correct date is returned
+                var from = testcase[yearFrom];
+                var rolledOverDate = Rollover.rollOverTimestamp(from, yearFrom, yearTo);
+                assert.strictEqual(rolledOverDate, testcase[yearTo]);
+                count++;
+            });
+        });
+
+        // Ensure we completed the expected number of test cases
+        assert.strictEqual(count, rolls.length * testData.testcases.length);
+    });
+
+    /**
+     * Verify dateInBetween returns the day between two dates
+     */
+    it('verify dateInBetween returns the day between two dates', function() {
+        var cases = [
+            {'a': [2014, 1, 5], 'b': [2014, 1, 6], 'expected': '2014-02-05T00:00:00+00:00'},
+            {'a': [2014, 1, 5], 'b': [2014, 1, 7], 'expected': '2014-02-06T00:00:00+00:00'},
+            {'a': [2014, 1, 5], 'b': [2014, 1, 10], 'expected': '2014-02-07T00:00:00+00:00'},
+            {'a': [2014, 1, 5], 'b': [2014, 1, 15], 'expected': '2014-02-10T00:00:00+00:00'}
+        ];
+
+        _.each(cases, function(testcase) {
+            var between = Rollover.dateInBetween(moment.utc(testcase.a), moment.utc(testcase.b));
```

Comment:
```
Note: Because this mirrors the native Date parameters, months, hours, minutes, seconds, and milliseconds are all zero indexed. Years and days of the month are 1 indexed.

This is often the cause of frustration, especially with months, so take note!
```

hehe